db/schema_tables.cc

```diff
@@ -974,23 +990,32 @@ static read_table_names_of_keyspace(distributed<service::storage_proxy>& proxy,
     }));
 }
 
-static table_id table_id_from_mutations(const schema_mutations& sm) {
+static std::optional<table_id> table_id_from_mutations(const schema_mutations& sm) {
     auto table_rs = query::result_set(sm.columnfamilies_mutation());
+    if (table_rs.empty()) {
+        return std::nullopt;
+    }
     query::result_set_row table_row = table_rs.row(0);
     return table_id(table_row.get_nonnull<utils::UUID>("id"));
 }
 
 static
 future<std::map<table_id, schema_mutations>>
-read_tables_for_keyspaces(distributed<service::storage_proxy>& proxy, const std::set<sstring>& keyspace_names, schema_ptr s)
+read_tables_for_keyspaces(distributed<service::storage_proxy>& proxy, const std::set<sstring>& keyspace_names, schema_ptr s,
+                          const std::unordered_map<sstring, std::optional<std::set<sstring>>>& tables_per_keyspace)
 {
     std::map<table_id, schema_mutations> result;
-    for (auto&& keyspace_name : keyspace_names) {
-        for (auto&& table_name : co_await read_table_names_of_keyspace(proxy, keyspace_name, s)) {
+    for (auto&& [keyspace_name, maybe_tables] : tables_per_keyspace) {
+        for (auto&& table_name : *maybe_tables) {
```

Comment:
No check for optional::has_value?

Reply:
It's guaranteed to be engaged here. We read the missing maybe_tables in the caller earlier.